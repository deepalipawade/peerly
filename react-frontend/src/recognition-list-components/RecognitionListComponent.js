import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

import RecognitionCardComponent from "recognition-list-components/RecognitionCardComponent";

const Wrapper = styled.div`
  border: 1px solid lightgrey;
  background: #ffffff 0% 0% no-repeat padding-box;
  border-radius: 36px;
  opacity: 1;
`;

const RecognitionListComponent = ({ recognitionList }) => {
  const getDateFormate = (timestamp) =>
    new Date(timestamp * 1000).toDateString();

  return (
    <Wrapper>
      {recognitionList.map((recognition, index) => (
        <RecognitionCardComponent
          key={index}
          givenByName={`${recognition.givenBy.first_name} ${recognition.givenBy.last_name}`}
          givenByImage={recognition.givenBy.profile_image_url}
          givenForName={`${recognition.givenFor.first_name} ${recognition.givenFor.last_name}`}
          givenForImage={recognition.givenFor.profile_image_url}
          givenAt={getDateFormate(recognition.given_at)}
          text={recognition.text}
          coreValue={recognition.coreValue.text}
          coreValueImage={recognition.coreValue.thumbnail}
        />
      ))}
    </Wrapper>
  );
};

RecognitionListComponent.propTypes = {
  recognitionList: PropTypes.array.isRequired,
};

export default React.memo(RecognitionListComponent);
