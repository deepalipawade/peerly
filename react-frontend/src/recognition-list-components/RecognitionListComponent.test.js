import React from "react";
import { render } from "@testing-library/react";
import RecognitionListComponent from "recognition-list-components/RecognitionListComponent";

describe("Recognition list Component test", () => {
  test("should equal snapshot", () => {
    const { asFragment } = render(
      <RecognitionListComponent recognitionList={[]} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
