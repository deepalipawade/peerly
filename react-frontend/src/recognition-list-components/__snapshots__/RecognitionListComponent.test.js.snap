// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recognition list Component test should equal snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-Axmtr cRMTYu"
  />
</DocumentFragment>
`;
